pykechain/models/widgets/widgets_manager.py

```diff
@@ -356,8 +367,10 @@ def add_supergrid_widget(
             }
         )
 
-        if parent_instance:
-            meta[AssociatedObjectId.PARENT_INSTANCE_ID] = parent_instance
+        if parent_instance_id:
+            meta[AssociatedObjectId.PARENT_INSTANCE_ID] = parent_instance_id
```

Comment:
I think the meta must be instantiated with an `AssociatedObjectId.PARENT_INSTANCE_ID` = None, when the `parent_part.category == Category.MODEL`. Hence on line 336, makde parent_isntance_id = None instead of str() and remove the if statement on line 370 will resolve this.

Reply:
parent instance id cannot be None, but it needs to be an UUID if it is provided

Reply:
Misunderstanding, parent instance id should be removed from meta if not provided